### PR TITLE
Refactor repository discovery Git command boundary

### PR DESCRIPTION
**Pull request by errfld**
Refs #188

## Summary
- Introduce an internal `GitClient` boundary for repository `rev-parse` interactions.
- Keep repository semantics (`find_git_root`, `resolve_git_path`) in `repository.rs` while moving process spawning/status/stderr handling behind typed command results.
- Add fake-client unit coverage so repository path behavior can be tested without spawning Git.

## Validation
- `cargo fmt --all -- --check`
- `cargo test -p git-smee-core repository -- --nocapture`
- `cargo test -p git-smee-cli doctor`
- `cargo clippy --workspace --all-targets --all-features -- -D warnings`
- `cargo test --workspace --all-targets --all-features`

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved Git repository detection and Git path resolution, making behavior more reliable in edge cases and when running outside a repository.
  * Error messages for Git path resolution now surface more useful details when resolution fails.
* **Tests**
  * Added broader unit coverage for repository discovery and Git path handling, improving confidence in path-related behavior.

### CODE REVIEW

File: crates/git-smee-core/src/repository.rs [hunk 1]
```diff
@@ -128,68 +132,161 @@ fn find_git_root_from_path(current_dir: &Path) -> Result<PathBuf, Error> {
...
+
+        if !output.status.success() {
+            return Ok(GitCommandResult::Failure(GitCommandFailure::from_output(
+                &output.stderr,
+                output.status,
+            )));
+        }
+
+        Ok(GitCommandResult::Success(output.stdout))
+    }
```
[the remark this answers is not in the document]
Comment (errfld) [reply crates/git-smee-core/src/repository.rs:249]: Addressed in 55d5319 by adding a shared run_git_command helper and routing bool/path/git-path rev-parse calls through it. Re-ran fmt, repository/doctor tests, clippy, and full workspace tests locally.
